aspnetcore/fundamentals/routing.md

```diff
@@ -5,275 +5,386 @@ description: Discover how ASP.NET Core routing is responsible for matching HTTP
 monikerRange: '>= aspnetcore-2.1'
 ms.author: riande
 ms.custom: mvc
-ms.date: 12/13/2019
+ms.date: 1/25/2020
 uid: fundamentals/routing
 ---
 # Routing in ASP.NET Core
 
-By [Ryan Nowak](https://github.com/rynowak), [Steve Smith](https://ardalis.com/), and [Rick Anderson](https://twitter.com/RickAndMSFT)
+By [Ryan Nowak](https://github.com/rynowak), [Kirk Larkin](https://twitter.com/serpent5), and [Rick Anderson](https://twitter.com/RickAndMSFT)
 
 ::: moniker range=">= aspnetcore-3.0"
 
-Routing is responsible for matching incoming HTTP requests and dispatching those requests to the applications executable endpoints. Endpoints are the apps' units of executable request-handling code. Endpoints are defined in the app and configured when the app starts. The matching process can extract values from the request's URL, and these values can then be used for request processing. Using endpoint information from the app, routing is also able to generate URLs that map to endpoints. 
+Routing is responsible for matching incoming HTTP requests and dispatching those requests to the apps executable endpoints. [Endpoints](#endpoint) are the apps' units of executable request-handling code. Endpoints are defined in the app and configured when the app starts. The endpoint matching process can extract values from the request's URL, and those values can be used for request processing. Using endpoint information from the app, routing is also able to generate URLs that map to endpoints.
 
 Apps can configure routing using:
 
 - Controllers
 - Razor Pages
 - SignalR
 - gRPC Services
-- Endpoint-enabled middleware (*routerware*) such as Health Checks
-- Using routing primitives directly.
+- Endpoint-enabled [middleware](xref:fundamentals/middleware/index) such as [Health Checks](xref:host-and-deploy/health-checks).
+- Delegates and Lambdas registered with routing.
 
```

Comment:
The final two bullets end with a `.`, but the first four don't. I dunno which is preferred, so just pointing it out in case it matters.